test/Microsoft.AspNet.Mvc.Core.Test/Rendering/HtmlHelperSelectTest.cs

```diff
@@ -858,6 +864,352 @@ public void ListBoxFor_WithUnreleatedExpression_GeneratesExpectedValue()
+            var htmlHelper = new TestHtmlHelper(metadataProvider);
+
+            // Act
+            var result = htmlHelper.GetSelectList<EnumWithFields>();
+
+            // Assert
+            Assert.Equal(metadata.ModelType, htmlHelper.Metadata.ModelType);
+
+            Assert.Same(htmlHelper.SelectListItems, result);            // No replacement of the underlying List
+            VerifySelectList(htmlHelper.CopiedSelectListItems, result); // No change to the (mutable) items
+        }
+
+        [Fact]
+        public void GetSelectListType_ThrowsWithFlagsEnum()
+        {
+            // Arrange
+            var metadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
+            var htmlHelper = new TestHtmlHelper(metadataProvider);
+
+            // Act & Assert
+            ExceptionAssert.ThrowsArgument(
+                () => htmlHelper.GetSelectList(typeof(EnumWithFlags)),
+                "enumType",
+                $"The type '{ typeof(EnumWithFlags).FullName }' is not supported.");
+        }
+
+        [Fact]
+        public void GetSelectListType_ThrowsWithNonEnum()
+        {
+            // Arrange
+            var metadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
+            var htmlHelper = new TestHtmlHelper(metadataProvider);
+
+            // Act & Assert
+            ExceptionAssert.ThrowsArgument(
+                () => htmlHelper.GetSelectList(typeof(StructWithFields)),
+                "enumType",
+                $"The type '{ typeof(StructWithFields).FullName }' is not supported.");
+        }
+
+        [Fact]
+        public void GetSelectListType_ThrowsWithNonStruct()
+        {
+            // Arrange
+            var metadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
+            var htmlHelper = new TestHtmlHelper(metadataProvider);
+
+            // Act & Assert
+            ExceptionAssert.ThrowsArgument(
+                () => htmlHelper.GetSelectList(typeof(ClassWithFields)),
+                "enumType",
+                $"The type '{ typeof(ClassWithFields).FullName }' is not supported.");
+        }
+
+        [Fact]
+        public void GetSelectListType_WrapsGetSelectListModelMetadata()
+        {
+            // Arrange
+            var metadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
+            var metadata = metadataProvider.GetMetadataForType(typeof(EnumWithFields));
+            var htmlHelper = new TestHtmlHelper(metadataProvider);
+
+            // Act
+            var result = htmlHelper.GetSelectList(typeof(EnumWithFields));
+
+            // Assert
+            Assert.Equal(metadata.ModelType, htmlHelper.Metadata.ModelType);
+
+            Assert.Same(htmlHelper.SelectListItems, result);            // No replacement of the underlying List
+            VerifySelectList(htmlHelper.CopiedSelectListItems, result); // No change to the (mutable) items
```

Comment:
I really don't understand what this is doing. This test (and `GetSelectListTEnum_WrapsGetSelectListModelMetadata`) just seems to me to verify the class you wrote to test this feature.